proposals/target-typed-new.md

```diff
@@ -46,54 +51,51 @@ The type of the expression would be inferred from the target-type which would be
 with the following exceptions:
 
 - **Enum types:** not all enum types contain the constant zero, so it should be desirable to use the explicit enum member.
-- **Delegate types:** if the type is inferrable, an anonymous function can already be used.
 - **Interface types:** this is a niche feature and it should be preferable to explicitly mention the type.
-- **Tuple types:** tuple literals should be used for such expressions.
 - **Array types:** arrays need a special syntax to provide the length.
+- **Struct default constructor**: this rules out all primitive types and most value types. If you wanted to use the default value of such types you could write `default` instead.
 
 All the other types that are not permitted in the *object_creation_expression* are excluded as well, for instance, pointer types.
 
-The default constructor for a value type (unless appeared explicitly in metadata) is excluded, which has the effect of excluding the primitive types and the default constructor for most value types. If you wanted to use the default value of such types you could write `default`.
-
-> **Open Issue:** What other types should be excluded?
+> **Open Issue:** should we allow delegates and tuples as the target-type?
 
-Note that any restriction on permitted types would raise the success rate of the overload resolution. For example, the following would successfully compile considering the restriction on the default constructor for value types.
+Although both types are constructible, if the type is inferrable, an anonymous function or a tuple literal can already be used.
 ```cs
-class C {}
-void M(C c) {}
-void M(int i) {}
+(int a, int b) t = new(1, 2); // "new" is redundant
+Action a = new(() => {}); // "new" is redundant
 
-M(new());
+(int a, int b) t = new(); // ruled out by "use of struct default constructor"
+Action a = new(); // no constructor found
+
+var x = new() == (1, 2); // ruled out by "use of struct default constructor"
+var x = new(1, 2) == (1, 2) // "new" is redundant
 ```
-Otherwise it would fail with an ambiguous call error.
 
-## Drawbacks
-[drawbacks]: #drawbacks
 
-Since we're relying on overload resolution of the target-type members, there is a high chance that any change in one of the participating members (target-type's constructors or call-site method overloads, if any) would result in breaking the compilation. For instance,
-```cs
-class Foo { }
-class Bar { }
+> **Open Issue:** should we allow `throw new()` with `Exception` as the target-type?
 
```

Comment:
It already is allowed by the current specification. Exception is a reference type, and the specification for the throw statement says that the expression is converted to Exception.